### PR TITLE
ERR: doc update for ParsingError

### LINKED ISSUE #22881: DOC: Fix and extend ParserError docstring
[`ParserError` says](https://github.com/pandas-dev/pandas/blob/8e51fd38607e94524b66ba15002e2502240baa32/pandas/errors/__init__.py#L35):
>Exception that is raised by an error encountered in `pd.read_csv`

however, it seems to be raised at least by html parsing, and perhaps other parsers as well: https://github.com/pandas-dev/pandas/blob/486e626bd3e7282b3da9993807f6246ef3ddda76/pandas/tests/io/test_html.py#L820-L829

As higher level point, it would be great if the list of potential runtime exceptions raised by the the various parsers were documented in the docs.

### PR DESCRIPTION
**Pull request by sudhirmohanraj**
Added documentation to Parse Error.

- [x] closes #22881 
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by sudhirmohanraj** [comment]
@cancan101 Can you review the documentation update?

File: pandas/errors/__init__.py [hunk 1]
```diff
@@ -32,6 +32,8 @@ class UnsortedIndexError(KeyError):
 class ParserError(ValueError):
     """
     Exception that is raised by an error encountered in `pd.read_csv`.
+
+    e.g. HTML Parsing and  malformed input csv files will raise this error.
```
Comment (jreback) [inline pandas/errors/__init__.py]: this is partially redudant

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/errors/__init__.py (hunk 1 above)
Comment (sudhirmohanraj) [reply pandas/errors/__init__.py]: Ok, I just looked at various flows that could raise this error. I will update it to just have the HTML parsing.

**Review by jreback — APPROVED** [review]

**Review by gfyoung — APPROVED** [review]
Hmm...this will address the issue, but I would like to do more of a rewrite along the lines of what is done for `DtypeWarning` after this.

**Comment by gfyoung** [comment]
Thanks @sudhirmohanraj !

**Pull request merged**